Sources/Asset.swift

```diff
@@ -246,4 +246,50 @@ extension Asset: ResourceQueryable {
                             }
         })
     }
+    
+    public func asyncFetch<ResourceType>(_ completion: @escaping (Result<ResourceType, Error>, ResponseType) -> Void) async
```

Comment:
withContinue function should be here instead of in the stack class. @vkalta .
Using async function in stack is unnecessary. instead use it directory in the specified classes

Reply:
The withContinue function is used in stack class and is used by other modules like content-type, entry and asset. If I write it in each module it will have too much of repetition of code.